staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -94,27 +97,26 @@ func getResourceHandler(scope RequestScope, getter getterFunc) http.HandlerFunc
 		ctx := scope.ContextFunc(req)
 		ctx = request.WithNamespace(ctx, namespace)
 
-		result, err := getter(ctx, name, req)
+		result, err := getter(ctx, name, req, trace)
 		if err != nil {
 			scope.err(err, w, req)
 			return
 		}
+		trace.Step("About to create self-link")
```

Comment:
As I wrote above, in lists it can take a lot, because there can be a lot of objects within list. Here you have exactly one object so it will be instantanous.